R/latent_lognormal.R

```diff
@@ -0,0 +1,84 @@
+#' Draws from the posterior predictive distribution
+#'
+#' See [`brms::posterior_predict`].
+#'
+#' @param i The index of the observation to predict
+#' @param prep The result of a call to [`brms::posterior_predict`]
+#' @family postprocess
+#' @autoglobal
+#' @export
+posterior_predict_latent_lognormal <- function(i, prep, ...) { # nolint: object_length_linter
+  mu <- brms::get_dpar(prep, "mu", i = i)
+  sigma <- brms::get_dpar(prep, "sigma", i = i)
+
+  wN <- prep$data$wN[i] # nolint: object_name_linter
+  woverlap <- prep$data$woverlap[i]
+  noverlap <- prep$data$noverlap[i]
+
+  # This assumes a certain prior on swindow_raw and pwindow_raw
+  # It would be better to extract these from prep or the model in some way?
+  swindow_raw <- runif(1, 0, 1)
+  pwindow_raw <- runif(1, 0, 1)
+
+  vreal3 <- prep$data$vreal3[i]
+  swindow <- vreal3 * swindow_raw
+
+  vreal2 <- prep$data$vreal2[i]
+  pwindow <- vreal2[noverlap] * pwindow_raw[noverlap]
+  if (wN) {
+    pwindow[woverlap] <- swindow[woverlap] * pwindow_raw[woverlap]
+  }
+
+  vreal1 <- prep$data$vreal1[i]
+  obs_t <- vreal1
+  obs_time <- obs_t - pwindow
+
+  d <- EnvStats::rlnormTrunc(
+    1,
+    meanlog = mu,
+    sdlog = sigma,
+    min = 0,
+    max = obs_time
```

Comment:
Think more about this (?)

Reply:
Look at `simulate_double_censored`

Reply:
`vreal2 <- prep$data$vreal2[i]` contains the information about the size of the window